service/frontend/adminHandler.go

```diff
@@ -117,12 +117,16 @@ func (adh *AdminHandler) RegisterHandler() {
 // Start starts the handler
 func (adh *AdminHandler) Start() {
 	// Start domain replication queue cleanup
-	adh.Resource.GetDomainReplicationQueue().Start()
+	if adh.config.EnableCleanupReplicationTask() {
+		adh.Resource.GetDomainReplicationQueue().Start()
+	}
 }
 
 // Stop stops the handler
 func (adh *AdminHandler) Stop() {
-	adh.Resource.GetDomainReplicationQueue().Stop()
+	if adh.config.EnableCleanupReplicationTask() {
+		adh.Resource.GetDomainReplicationQueue().Stop()
+	}
```

Comment:
The value for this dynamicconfig may and will change when the service is running, so it's possible only `Start()` is called or only `Stop()` is called. Ideally the config value should be get once and stored somewhere on service startup.

I checked the implementation for domain replication queue, it seems to be ok if only `Stop()` is called, not sure about what will happen if only `Start()` is called. But if you're relying on this fact, probably add some comments in the code.

BTW, there's a copy & paste error on `domainReplicationQueue.go` L104.

Reply:
Good point. So I updated it to always call stop() and check the flag before calling start()

Reply:
Please also add some comments explaining that it's ok to call `Stop()` without calling `Start()`.